src/app.js

```diff
@@ -43,29 +43,36 @@ app.get('/help', (req, res) => {
 })
 
 app.get('/weather', (req, res) => {
-    if (!req.query.address) {
+    if (!req.query.address && !req.query.lat && !req.query.lon) {
         return res.send({
-            error: 'No address provided!'
+            error: 'No location provided!'
         })
     }
-    geocode.geocode(req.query.address, (error, { latitude, longitude, location } = {}) => {
-        if (error) {
-            return res.send({ error })
-        }
-        forecast.forecast(latitude, longitude, (error, { description, temperature, feelslike, icon }) => {
+
+    if (req.query.address) {
+        geocode.geocode(req.query.address, (error, { latitude, longitude } = {}) => {
             if (error) {
                 return res.send({ error })
             }
-            res.send({
-                description,
-                temperature,
-                feelslike,
-                icon,
-                location,
-                address: req.query.address
+            forecast(latitude, longitude, async(error, data) => {
+                try {
+                    const result = await forecastErrCheck(error, data)
+                    res.status(200).send(result)
+                } catch (error) {
+                    console.log(error)
+                }
             })
         })
-    })
+    } else {
+        forecast(req.query.lat, req.query.lon, async(error, data) => {
+            try {
+                const result = await forecastErrCheck(error, data)
+                res.status(200).send(result)
+            } catch (error) {
+                console.log(error)
```

Comment:
client gets timeout in that case

Reply:
Could you please clarify what you mean?
Maybe the issue is outdated?